### PR TITLE
Feature/add new query amplify app access token rule

### LINKED ISSUE #971: Add amplify_app_access_token_rule query for AWS Cloud Formation
### Platform
AWS Cloud Formation

### Provider
AWS

### Description
Amplify App Access Token must not be in a plain text string or referenced in a parameter as a default value.

### PR DESCRIPTION
**Pull request by cataraujo190**
Description: Amplify App Access Token must not be in a plain text string or referenced in a parameter as a default value.

Close #971

### CODE REVIEW

File: assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Resources:
+     NewAmpApp-2:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml:2]: Remove `-`
Linter message: `Resource NewAmpApp-2 has an invalid name.  Name has to be alphanumeric.`

File: assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            GenerateStringKey: 'password'
+            PasswordLength: 16
+            ExcludeCharacters: '"@/\'
+---
+Parameters:
+  ParentAccessToken:
+    Description: 'Access Token'
+    Type: String
+Resources:
+  NewAmpApp-1:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml:29]: Remove `-`
Linter message: `Resource NewAmpApp-1 has an invalid name.  Name has to be alphanumeric.`

File: assets/queries/cloudFormation/amplify_app_access_token_rule/test/positive.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      OauthToken: String
+      Repository: String
+---
+Parameters:
+  ParentAccessToken:
+    Description: 'Access Token'
+    Type: String
+    Default: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1MTYyMzkwMjJ9.tbDepxpstvGdW8TC3G8zg4B6rUYAOvfzdceoH48wgRQ'
+Resources:
+  NewAmpApp-1:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/test/positive.yaml:21]: Remove `-`
Linter message: `Resource NewAmpApp-1 has an invalid name.  Name has to be alphanumeric.`

File: assets/queries/cloudFormation/amplify_app_access_token_rule/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      Name: NewAmpApp
+      OauthToken: String
+      Repository: String
+---
+Parameters:
+  ParentUserToken:
+    Description: 'UserToken'
+    Type: String
+Resources:
+  NewAmpApp-1:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/test/positive.yaml:39]: Remove `-`
Linter message: `Resource NewAmpApp-1 has an invalid name.  Name has to be alphanumeric.`

File: assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    Properties:
+      AccessToken: !Ref ParentAccessToken
+      BuildSpec: String
+      CustomHeaders: String
+      Description: String
+      EnableBranchAutoDeletion: true
+      IAMServiceRole: String
+      Name: NewAmpApp
+      OauthToken: String
+      Repository: String
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/test/negative.yaml:59]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add EOF newline

File: assets/queries/cloudFormation/amplify_app_access_token_rule/metadata.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "amplify_app_access_token_rule",
+  "queryName": "Amplify App Access Token Rule",
+  "severity": "MEDIUM",
+  "category": null,
+  "descriptionText": "Amplify App Access Token must not be in a plain text string or referenced in a parameter as a default value.",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-amplify-app.html"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add EOF newline

File: assets/queries/cloudFormation/amplify_app_access_token_rule/query.rego [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.Properties.AccessToken", [key]),
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue": sprintf("Resources.%s.Properties.AccessToken must not be in plain text string",[key]),
+                "keyActualValue": 	sprintf("Resources.%s.Properties.AccessToken must be defined as a parameter or have a secret manager referenced",[key])
+              }
+}
+
+
+hasSecretManager(str, document) {
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/query.rego:84]: remove duplicate function `hasSecretManager`

File: assets/queries/cloudFormation/amplify_app_access_token_rule/query.rego [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+  document := input.document[i]
+  resource := document.Resources[key]
+  resource.Type == "AWS::Amplify::App"
+
+  properties := resource.Properties
+  paramName  := properties.AccessToken
+  defaultToken := document.Parameters[paramName].Default
+  count(defaultToken) > 50
+  regex.match(`^[A-Za-z0-9-_=]+\.[A-Za-z0-9-_=]+\.?[A-Za-z0-9-_.+/=]*$`,defaultToken) #Access Token is a JWT token from following docs: https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#amazon-cognito-user-pools-using-the-access-token
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/amplify_app_access_token_rule/query.rego:14]: To reduce the number of columns in this file, place this inline comment on the previous line.

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Almost good.

**Review by rogeriopeixotocx — DISMISSED** [review]
LGTM

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**